scripts/ci-equivalent-check.sh

```
@@ -0,0 +1,113 @@
+# dependency-guard or non-Android targets because the resolution tree differs.
+#
+# Usage:
+#   bash scripts/ci-equivalent-check.sh
+#
+# Exit code:
+#   0 = all gates green (safe to push)
+#   1 = at least one gate failed (DO NOT push without fixing)
+
+set -uo pipefail
+
+cd "$(dirname "$0")/.."
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)